usr/src/uts/common/io/scsi/adapters/smartpqi/smartpqi_sis.c

```diff
@@ -0,0 +1,251 @@
+/*
+ * This file and its contents are supplied under the terms of the
+ * Common Development and Distribution License ("CDDL"), version 1.0.
+ * You may only use this file in accordance with the terms of version
+ * 1.0 of the CDDL.
+ *
+ * A full copy of the text of the CDDL should have accompanied this
+ * source.  A copy of the CDDL is also available via the Internet at
+ * http://www.illumos.org/license/CDDL.
+ */
+
+/*
+ * Copyright 2018 Nexenta Systems, Inc.
+ */
+
+/*
+ * A few simple method to access controller when it's in the base mode.
+ */
+#include <smartpqi.h>
+
+/* ---- legacy SIS interface commands ---- */
+#define	SIS_CMD_GET_ADAPTER_PROPERTIES	0x19
+#define	SIS_CMD_INIT_BASE_STRUCT_ADDRESS	0x1b
+#define	SIS_CMD_GET_PQI_CAPABILITIES		0x3000
+
+/* ---- used with SIS_CMD_GET_ADAPTER_PROPERTIES command ---- */
+#define	SIS_EXTENDED_PROPERTIES_SUPPORTED	0x800000
+#define	SIS_SMARTARRAY_FEATURES_SUPPORTED	0x2
+#define	SIS_PQI_MODE_SUPPORTED			0x4
+#define	SIS_REQUIRED_EXTENDED_PROPERTIES	\
+	(SIS_SMARTARRAY_FEATURES_SUPPORTED | SIS_PQI_MODE_SUPPORTED)
+
+#pragma pack(1)
```

Comment:
The structures below are already going to be packed, so you can dispense with the compiler-specific #pragma